src/app/qgsrelationadddlg.cpp

```diff
@@ -24,45 +25,41 @@ QgsRelationAddDlg::QgsRelationAddDlg( QWidget *parent )
 {
   setupUi( this );
 
-  mTxtRelationId->setPlaceholderText( tr( "[Generated automatically]" ) );
-  checkDefinitionValid();
+  connect( mCbxReferencingLayer, &QgsMapLayerComboBox::layerChanged, this, [=]( QgsMapLayer * layer ) { mCbxReferencingField->setLayer( layer ); }
```

Comment:
`QgsMapLayer* layer`

Everything else is just plain evil. I just can't get qt creator to reliably respect this setting. I wonder if astyle could handle this...